__tests__/app.test.js

```diff
@@ -265,5 +261,57 @@ describe('GET /api/articles/:article_id/comments', () => {
         expect(body.message).toEqual("Resource not found");
       });
   });
-})
+});
+
+describe.only("POST /api/articles/:article_id/comments", () => {
+  test("Request body takes a user and body and returns the posted comment", () => {
+    const newComment = {
+      username: "icellusedkars",
+      body: "This is an amazing comment.",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body).toEqual({
+          comment_id: 19,
+          body: "This is an amazing comment.",
+          article_id: 2,
+          author: "icellusedkars",
+          votes: 0,
+          created_at: expect.any(String),
+        });
+      });
+  });
+
+  test("400 - missing required fields", () => {
+    const newComment = {
+      username: "icellusedkars",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .expect(400)
+      .then(({ body }) => {
+        console.log(body.message);
+        expect(body.message).toEqual(
+          "Please ensure valid body and username are entered"
+        );
+      });
+  });
 
+  test("400 - username does not exist", () => {
+    const newComment = {
+      username: "emmaMadd",
+      body: "This is an amazing comment.",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.message).toEqual(
+          "Please ensure valid body and username are entered"
+        );
+      });
+  });
+});
```

Comment:
Should also test for:
- [ ] Status 404, non existent ID, e.g. 0 or 9999
- [ ] Status 201, ignores unnecessary properties

These have been tested for in other areas of functionality, but there's no guarantee that they would work on this route.